keras/models.py

```diff
@@ -32,6 +32,90 @@
+    When you call save_mxnet_model, these information will be printed.
+    You can change the first dimension of data_shapes to match batch size for inference.
+
+    Below sample code shows how to load a Keras-MXNet pre-trained model saved with 'prefix',
+    input name as '/dense_1_input1' and inference batch size of 1 and data shape of 784.
+
+    ```python
+    >>> import mxnet as mx
+    >>> sym, arg_params, aux_params = mx.model.load_checkpoint(prefix, epoch=0)
+    ##  data_names and data_shapes will be print
+    >>> mod = mx.mod.Module(symbol=sym, data_names=['/dense_1_input1'], \
+                            context=mx.cpu(), label_names=None)
+    >>> mod.bind(for_training=False, data_shapes=[('/dense_1_input1', (1,784))], \
+                 label_shapes=mod._label_shapes)
+    >>> mod.set_params(arg_params, aux_params, allow_missing=True)
+    ##  Prepare data iterator
+    >>> data_iter = mx.io.NDArrayIter(data, label=None, batch_size=1)
+    ##  Predict
+    >>> result = mod.predict(data_iter)
+    ```
+
+    # Arguments
+        model: Keras model instance to be saved as MXNet model.
+        prefix: Prefix name of the saved Model (symbol and params) files.
+                Model will be saved as '<prefix>-symbol.json' and '<prefix>-<epoch>.params'.
+        epoch: (Optional) Tag the params file with epoch of the model being saved. Default is 0.
+               Model params file is saved as '<prefix>-<epoch>.params' or '<prefix>-0000.params' by default.
+
+    # Returns
+        data_names, data_shapes
+    """
+    assert model is not None, 'MXNet Backend: Invalid state. Model cannot be None.'
+
+    # Handle Sequential Model / Functional API Model Case
+    if isinstance(model, Sequential):
+        mxnet_model = model.model
+    elif isinstance(model, Model):
+        mxnet_model = model
+    else:
+        raise ValueError('MXNet Backend: Invalid model type. Supported Models - Sequential, Model.')
+
+    assert mxnet_model is not None, 'MXNet Backend: Invalid state. MXNet Model cannot be None.'
+
+    if not mxnet_model.compiled:
+        raise ValueError('MXNet Backend: Model is not compiled. Cannot save the MXNet model!')
+
+    # Saving MXNet model for Inference in native MXNet engine.
+    symbol = mxnet_model._pred_mxnet_symbol
+    module = mxnet_model._module
+
+    assert symbol is not None, 'MXNet Backend: Invalid state. MXNet Symbol cannot be None.'
+    assert module is not None, 'MXNet Backend: Invalid state. MXNet Module cannot be None.'
+
+    # Get Module Input data_names and data_shapes.
+    # This info will be useful for users to easily bind the exported model in MXNet.
+    pred_module = module._buckets['pred']
+    data_names = pred_module.data_names
+    data_shapes = pred_module.data_shapes
+
+    symbol_fname = '%s-symbol.json' % prefix
+    params_fname = '%s-%04d.params' % (prefix, epoch)
+    symbol.save(symbol_fname)
```

Comment:
does it make sense to return the symbol name and the params too? In case you need them during the session? Writing to disk and reading back may not be optimal

Reply:
Good suggestion. Probably a different API for that - 'keras.models.get_mxnet_model(model)' ?